packages/drivers/odsp-driver-definitions/src/resolvedUrl.ts

```diff
@@ -12,14 +12,49 @@ export interface IOdspUrlParts {
 }
 
 /**
+ * @deprecated Use ShareLink type instead.
  * Type of shareLink requested/created when creating the file for the first time.
- * At the time of adding this comment (Sept/2021) ODSP only supports creation of CSL links
- * when provided as a request parameter with the /snapshot api call.
- * In future, we can add more types here.
 */
 export enum ShareLinkTypes {
     csl = "csl",
 }
+
+/**
+ * Sharing scope of the share links created for a file.
+*/
+export enum ShareLinkScope {
+    organization = "organization",
+    users = "users",
+    anonymous = "anonymous",
+    default = "default",
+}
+
+/**
+ * View/edit permission role for a sharing link.
+ */
+export enum ShareLinkRole {
+    view = "view",
+    edit = "edit",
+}
+
+/**
+ * Defines the sharing scope and permissions for a share link created for a file.
+*/
+export interface ShareLink {
```

Comment:
ShareLink does not look like right name. The context does not look like link to me.
Maybe ShareLinkInfo? or ShareLinkContext? Please give it right name that describes what it is

Reply:
Renamed `ShareLink` to `SharingLinkKind` and added code comments. SharingLinkKind would represent the kind of sharing link that is being requested to be created along with the creation of a file.